src/Storages/StorageDistributed.cpp

```diff
@@ -866,7 +866,7 @@ void StorageDistributed::flushClusterNodesAllData()
 
 void StorageDistributed::rename(const String & new_path_to_table_data, const StorageID & new_table_id)
 {
-    if (!relative_data_path.empty())
+    if (!relative_data_path.empty() && relative_data_path != new_path_to_table_data)
```

Comment:
When does it happen, that `relative_data_path` is equal to `new_path_to_table_data`? Seems like it may happen only in case of logical error, because `InterpreterRenameQuery` checks that table with new name does not exist and `DatabaseAtomic` does not call `IStorage::rename(...)` for persistent paths.

Reply:
It cannot happen... I made this change only looking at the code locally.
Maybe considered as "defensive programming" - the function does not know that it is never called with identical arguments.

Reply:
Then maybe add `assert` or throw `LOGICAL_ERROR`?

Reply:
Ok.